packages/graphql/src/graphql/directives/type-dependant-directives/static-definitions.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { astFromEnumType, astFromInputObjectType } from "@graphql-tools/utils";
+import type {
+    ObjectTypeDefinitionNode,
+    EnumTypeDefinitionNode,
+    InputObjectTypeDefinitionNode,
+    GraphQLInputObjectType,
+} from "graphql";
+import { GraphQLEnumType, GraphQLSchema } from "graphql";
+import { SchemaComposer } from "graphql-compose";
+import getWhereFields from "../../../schema/get-where-fields";
+import { getJwtFields } from "./jwt-payload";
+
+export const AUTHORIZATION_VALIDATE_STAGE = new GraphQLEnumType({
+    name: "AuthorizationValidateStage",
+    values: { BEFORE: { value: "BEFORE" }, AFTER: { value: "AFTER" } },
```

Comment:
This might be a silly question but which operations does this affect again? As far as I can tell only `UPDATE` should need validation before and after the operation which makes me think we just need to have `BEFORE_UPDATE` & `AFTER_UPDATE` as operations instead